components/application-authenticators/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/servlet/IWAServelet.java

```diff
@@ -228,95 +123,41 @@ private boolean doFilterPrincipal(HttpServletRequest request) throws IOException
      */
     private void sendUnauthorized(HttpServletResponse response, boolean close) {
         try {
-            IWAServiceDataHolder.getInstance().getProviders().sendUnauthorized(response);
             if (close) {
-                response.setHeader(IWAConstants.HTTP_CONNECTION_HEADER, IWAConstants.CONNECTION_CLOSE);
+                response.setHeader(IWAConstants.AUTHENTICATE_HEADER, IWAConstants.NEGOTIATE_HEADER);
+                response.addHeader(IWAConstants.HTTP_CONNECTION_HEADER, IWAConstants.CONNECTION_CLOSE);
             } else {
-                response.setHeader(IWAConstants.HTTP_CONNECTION_HEADER, IWAConstants.CONNECTION_KEEP_ALIVE);
+                response.setHeader(IWAConstants.AUTHENTICATE_HEADER, IWAConstants.NEGOTIATE_HEADER);
+                response.addHeader(IWAConstants.HTTP_CONNECTION_HEADER, IWAConstants.CONNECTION_KEEP_ALIVE);
             }
-            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
             response.flushBuffer();
         } catch (IOException e) {
             log.error("Error when sending unauthorized response." + e);
```

Comment:
Change this also to log.error("Error when sending unauthorized response.", e);  even this is not something added by you